cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,613 @@
+                self.filter_by_dropdown.item_select(type)
+            except NoSuchElementException:
+                logger.info("{} not present in filter dropdown!".format(type))
+            self.search_box.fill(name)
+            self.after_fill(was_change=True)
+
+        def select_by_name(self, vm_name):
+            self.filter("VM Name", vm_name)
+            vms_selected = []
+            for row in self.table.rows():
+                if vm_name in row.read()["VM Name"]:
+                    row.select.fill(True)
+                    vms_selected.append(row.read()["VM Name"])
+            return vms_selected
+
+    @View.nested
+    class advanced(View):  # noqa
+        pre_playbook = BootstrapSelect("preMigrationPlaybook")
+        post_playbook = BootstrapSelect("postMigrationPlaybook")
+        pre_checkbox = Text(locator='.//input[contains(@id, "pre_migration_select_all")]')
+        post_checkbox = Text(locator='.//input[contains(@id, "post_migration_select_all")]')
+
+        @property
+        def is_displayed(self):
+            return self.pre_playbook.is_displayed
+
+    @View.nested
+    class schedule(View):  # noqa
+        create = Button("Create")
+        run_migration = RadioGroup('.//div[contains(@id,"migration_plan_choice_radio")]')
+
+        @property
+        def is_displayed(self):
+            return self.run_migration.is_displayed
+
+    @View.nested
+    class results(View):  # noqa
+        close = Button("Close")
+        msg = Text('.//h3[contains(@id,"migration-plan-results-message")]')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+class MigrationPlanRequestDetailsView(View):
+    migration_request_details_list = MigrationPlanRequestDetailsList("plan-request-details-list")
+    sort_type = SelectorDropdown("id", "sortTypeMenu")
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu")
+    filter_by_status_dropdown = SelectorDropdown("id", "filterCategoryMenu")
+    sort_by_dropdown = SelectorDropdown("id", "sortTypeMenu")
+
+    @property
+    def is_displayed(self):
+        return self.migration_request_details_list.is_displayed
+
+    def filter_by_vm_name(self, vm_name):
+        """Enter VM Name in search box and hit ENTER to filter the list of VMs.
+
+        Args:
+            vm_name(str): Takes VM Name as arg.
+        """
+        try:
+            self.filter_by_dropdown.item_select("VM Name")
+        except NoSuchElementException:
+            logger.info(
+                "filter_by_dropdown not present, "
+                "migration plan may not have started yet.Ignoring."
+            )
+        self.search_box.fill(vm_name)
```

Comment:
Do you have the send the enter key here?

Reply:
Yes , done

Reply:
Will the search box be displayed when `NoSuchElementException` occurred when trying to use the dropdown before this?